perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/PerunBlImpl.java

```diff
@@ -153,6 +171,34 @@ public PerunSession getPerunSession(PerunPrincipal principal, PerunClient client
 		return perunSession;
 	}
 
+	/**
+	 * Iteratively searches through additional identifiers trying to find UES with the same identifier.
+	 * Returns first found UES or null when no matching UES found.
+	 *
+	 * @param principal PerunPrincipal which contains additionalIdentifiers
+	 * @param internalSession PerunSession to retrieve UserExtSource
+	 * @return UserExtSource found using additionalIdentifiers or null
+	 * @throws InternalErrorException When attribute is not defined in Perun
+	 */
+	public UserExtSource getUserExtSourceFromMultipleIdentifiers(PerunPrincipal principal, PerunSession internalSession) throws InternalErrorException {
+		String additionalIdentifiers = principal.getAdditionalInformations().get(additionalIdentifiersAttributeName);
+		UserExtSource ues = null;
+		for(String identifier : additionalIdentifiers.split(multivalueAttributeSeparatorRegExp)) {
+			try {
+				ues = usersManagerBl.getUserExtSourceByUniqueAttributeValue(internalSession, additionalIdentifiersPerunAttributeName, identifier);
+				log.debug("UserExtSource found using additional identifiers: " + ues);
+				break;
+			} catch(UserExtSourceNotExistsException ex) {
+				//try to find user ext source using different identifier in the next iteration of for cycle
+			} catch(AttributeNotExistsException ex) {
+				String errorMessage = "Mandatory attribute is not defined: ".concat(additionalIdentifiersPerunAttributeName);
+				log.error(errorMessage);
+				throw new InternalErrorException(errorMessage, ex);
+			}
+		}
+		return ues;
+	}
```

Comment:
It is really necessary to have this method twice in code?

Reply:
Absolutely not, this was result of some work on the method extraction discussed earlier and I made them exactly the same on purpose to emphasise that only one can be used for both situations. The same goes for those constants from next comment and my question is: where should they be located at and which way should I use to "import" them?

Reply:
There are two possible location (in my opinion). Because you have session, you can get perunBl from it, so one place where you can add it is to Utils.java. But much better place probably will be UsersManagerBlImpl.java, cause all methods returning object UserExtSource are there.